src/SWP/Bundle/ContentBundle/Loader/ArticleMediaLoader.php

```diff
@@ -0,0 +1,144 @@
+<?php
+
+/**
+ * This file is part of the Superdesk Web Publisher Content Bundle.
+ *
+ * Copyright 2015 Sourcefabric z.u. and contributors.
+ *
+ * For the full copyright and license information, please see the
+ * AUTHORS and LICENSE files distributed with this source code.
+ *
+ * @copyright 2015 Sourcefabric z.ú.
+ * @license http://www.superdesk.org/license
+ */
+namespace SWP\Bundle\ContentBundle\Loader;
+
+use SWP\Component\TemplatesSystem\Gimme\Context\Context;
+use SWP\Component\TemplatesSystem\Gimme\Factory\MetaFactory;
+use SWP\Component\TemplatesSystem\Gimme\Loader\LoaderInterface;
+use SWP\Component\TemplatesSystem\Gimme\Meta\Meta;
+use Symfony\Cmf\Bundle\CoreBundle\PublishWorkflow\PublishWorkflowChecker;
+use Doctrine\ODM\PHPCR\DocumentManager;
+use Doctrine\Common\Cache\CacheProvider;
+use SWP\Component\MultiTenancy\PathBuilder\TenantAwarePathBuilderInterface;
+
+class ArticleMediaLoader implements LoaderInterface
+{
+    /**
+     * @var PublishWorkflowChecker
+     */
+    protected $publishWorkflowChecker;
+
+    /**
+     * @var DocumentManager
+     */
+    protected $dm;
+
+    /**
+     * @var string
+     */
+    protected $configurationPath;
+
+    /**
+     * @var CacheProvider
+     */
+    protected $metadataCache;
+
+    /**
+     * @var TenantAwarePathBuilderInterface
+     */
+    protected $pathBuilder;
+
+
+    /**
+     * @var MetaFactory
+     */
+    protected $metaFactory;
+
+    /**
+     * @var Context
+     */
+    protected $context;
+
+    /**
+     * ArticleMediaLoader constructor.
+     *
+     * @param PublishWorkflowChecker          $publishWorkflowChecker
+     * @param DocumentManager                 $dm
+     * @param                                 $configurationPath
+     * @param CacheProvider                   $metadataCache
+     * @param TenantAwarePathBuilderInterface $pathBuilder
+     * @param MetaFactory                     $metaFactory
+     * @param Context                         $context
+     */
+    public function __construct(
```

Comment:
this looks to me like to many responsibilities, what about reducing number of passed parameters by splitting the code into more classes?